crates/edr_provider/src/data.rs

```diff
@@ -584,16 +584,16 @@ impl<LoggerErrorT: Debug> ProviderData<LoggerErrorT> {
     pub fn debug_trace_call(
         &mut self,
         transaction: ExecutableTransaction,
-        block_spec: Option<&BlockSpec>,
+        block_spec: &BlockSpec,
         trace_config: DebugTraceConfig,
     ) -> Result<DebugTraceResult, ProviderError<LoggerErrorT>> {
-        let cfg_env = self.create_evm_config(block_spec)?;
+        let cfg_env = self.create_evm_config(Some(block_spec))?;
```

Comment:
Nice to have: since this mistake was made before, it would be good to add documentation to `create_evm_config` that reads something along the lines of:

```rust
/// Creates an configuration, taking into the hardfork at the provided `BlockSpec`. If none is provided, assumes the hardfork for newly mined blocks.
```

Reply:
Thanks good idea, done in 780d39a4f6c6c6069d0482edae1e330927b96b6e